src/components/Map/hooks/useMarkers.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-loop-func */
+/* eslint-disable no-unused-vars */
+import { useRef, useEffect, useState, useMemo, useCallback } from 'react';
+import L from 'leaflet';
+import 'leaflet.markercluster';
+
+import { categories } from 'static/categories';
+import { getMarkerIcon } from 'services/marker';
+import useHighlight from './useHighlight';
+
+export const showDeviceInfo = (event, marker, onClick, highlight) => {
+  highlight(event.sourceTarget);
+  onClick(marker);
+};
+
+export const showAreaInfo = (event, onClick, highlight) => {
+  highlight(event.layer)
+  onClick(event.sourceTarget.feature);
+};
+
+const useMarkers = map => {
+  const markerGroupRef = useRef(null);
+  const layerListRef = useRef({});
```

Comment:
Looks like you're using this as component state. Shouldn't this be a `useReducer`?